### PR TITLE
[Wsign-compare] Close Wno-error=sign_compare

### PR DESCRIPTION
**Pull request by Li-fAngyU**
### PR types

Others 
### PR changes

Others 
### Describe

 Close Wno-error=sign_compare
- Compiler Warning tracking issue: https://github.com/PaddlePaddle/Paddle/issues/47143

### CODE REVIEW

**Review by luotao1 — APPROVED** [review]
LGTM

**Comment by luotao1** [comment]
我下载了这个PR流水线的PR-CI-Coverage和PR-CI-Build。Coverage的编译会多一个`deprecated-declaration`的warning，可以考虑后续也去掉。
```
$ grep warning: coverage.log |grep "\[\-W" |grep -v party | awk '{print $NF}'|sort|uniq -c|sort -nr
  21 [-Wdeprecated-declarations]
   5 [-Wterminate]
   4 [-Wunused-variable]
   4 [-Wunused-local-typedefs]
   2 [-Wunknown-pragmas]
$ grep warning: build.log |grep "\[\-W" |grep -v party | awk '{print $NF}'|sort|uniq -c|sort -nr
   6 [-Wterminate]
   4 [-Wunused-variable]
   4 [-Wunused-local-typedefs]
   2 [-Wunknown-pragmas]
```

**Comment by luotao1** [comment]
后续可以在PR描述里都加上tracking issue

File: paddle/fluid/distributed/CMakeLists.txt [hunk 1]
```diff
@@ -36,7 +36,7 @@ endif()
 proto_library(ps_framework_proto SRCS the_one_ps.proto)
 
 set(DISTRIBUTE_COMPILE_FLAGS
-    "-Wno-error=unused-value -Wno-non-virtual-dtor -Wno-error=non-virtual-dtor -Wno-error=delete-non-virtual-dtor -Wno-error=sign-compare -Wno-error=unused-variable -Wno-error=return-type -Wno-error=unused-but-set-variable -Wno-error=unknown-pragmas -Wno-error=parentheses -Wno-error=unused-result"
```
Comment (Li-fAngyU) [inline paddle/fluid/distributed/CMakeLists.txt:39]: fluid/distributed 下的CMakeLists里的Wno-error=sign-compare没有去除掉。
去除掉的话会触发第三方库的sign-compare error，[报错信息](https://xly.bce.baidu.com/paddlepaddle/paddle/newipipe/detail/6913358/job/19467796)

**Comment by Li-fAngyU** [comment]
> 我下载了这个PR流水线的PR-CI-Coverage和PR-CI-Build。Coverage的编译会多一个`deprecated-declaration`的warning，可以考虑后续也去掉。
> 
> ```
> $ grep warning: coverage.log |grep "\[\-W" |grep -v party | awk '{print $NF}'|sort|uniq -c|sort -nr
>   21 [-Wdeprecated-declarations]
>    5 [-Wterminate]
>    4 [-Wunused-variable]
>    4 [-Wunused-local-typedefs]
>    2 [-Wunknown-pragmas]
> $ grep warning: build.log |grep "\[\-W" |grep -v party | awk '{print $NF}'|sort|uniq -c|sort -nr
>    6 [-Wterminate]
>    4 [-Wunused-variable]
>    4 [-Wunused-local-typedefs]
>    2 [-Wunknown-pragmas]
> ```

好的

File: paddle/fluid/distributed/CMakeLists.txt (hunk 1 above)
Comment (luotao1) [reply paddle/fluid/distributed/CMakeLists.txt:39]: 报错信息：
```
/workspace/Paddle/paddle/fluid/distributed/test/brpc_utils_test.cc:119:3:   required from here
2022-10-19 16:44:33 /workspace/Paddle/build/third_party/install/gtest/include/gtest/gtest.h:1444:11: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
2022-10-19 16:44:33    if (lhs == rhs) {
```
https://github.com/PaddlePaddle/Paddle/blob/d6208aad900da90e9b5036b568d53a14f6e7ece4/paddle/fluid/distributed/test/brpc_utils_test.cc#L119
改成 `EXPECT_EQ(slr->rows().size(), 564UL);` 就不会有这个warning了，可以尝试去掉fluid/distributed 下的CMakeLists里的Wno-error=sign-compare

@Li-fAngyU

Comment (luotao1) [reply paddle/fluid/distributed/CMakeLists.txt:39]: 搜了下还有两处 https://github.com/PaddlePaddle/Paddle/search?q=sign-compare

Comment (Li-fAngyU) [reply paddle/fluid/distributed/CMakeLists.txt:39]: 好的我修改一下

**Pull request merged**